packages/flutter_tools/lib/src/analyze_project_validator.dart

```diff
@@ -0,0 +1,17 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'analyze_project.dart';
+import 'project.dart';
+
+abstract class ProjectValidator {
```

Comment:
Can you add dartdoc's for all the new classes you add, and any non-obvious methods? (for example `.supportsProject` is self-explanatory)